### PR TITLE
compileall: update for py39

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: stdlib/3/compileall.pyi [hunk 1]
```diff
@@ -34,11 +40,29 @@ if sys.version_info >= (3, 7):
         legacy: bool = ...,
         optimize: int = ...,
         invalidation_mode: Optional[PycInvalidationMode] = ...,
+        *,
+        stripdir: Optional[_Path] = ...,
```
Comment (JelleZijlstra) [inline stdlib/3/compileall.pyi:44]: This must actually be a string because it does `stripdir.split(os.path.sep)`

File: stdlib/3/compileall.pyi [hunk 2]
```diff
@@ -34,11 +40,29 @@ if sys.version_info >= (3, 7):
...
+        prependdir: Optional[_Path] = ...,
+        limit_sl_dest: Optional[_Path] = ...,
```
Comment (JelleZijlstra) [inline stdlib/3/compileall.pyi:46]: But these two really can be PathLike

File: stdlib/3/compileall.pyi (hunk 1 above)
Comment (hauntsaninja) [reply stdlib/3/compileall.pyi:44]: Great catch, contradicts the documentation. I filed https://bugs.python.org/issue40447

Comment (JelleZijlstra) [reply stdlib/3/compileall.pyi:44]: Since the bug probably won't be fixed very soon, can you just change the stub for now?

**Comment by hauntsaninja** [comment]
Updated the stub to reflect today's Python 3.9. Looking closer at this I think there are some issues here with `_Path` too. We can run into issues with bytes, since os.path.sep is a str + we probably need some typevars. Pretty surprising how explicit the documentation is, given how problematic that actual code is.
> The stripdir, prependdir and limit_sl_dest arguments correspond to the -s, -p and -e options described above. They may be specified as str, bytes or os.PathLike.

**Pull request merged**